src/py/flwr/client/app.py

```diff
@@ -396,13 +399,12 @@ def _on_backoff(retry_state: RetryState) -> None:
                     )
                 else:
                     # Call create_node fn to register node
-                    node_id: Optional[int] = (  # pylint: disable=assignment-from-none
-                        create_node()
-                    )  # pylint: disable=not-callable
-                    if node_id is None:
+                    # and store node_id in state
+                    state.set_node_id(create_node())
+                    if not state.get_node_id():
                         raise ValueError("Node registration failed")
                     run_info_store = DeprecatedRunInfoStore(
-                        node_id=node_id,
+                        node_id=cast(int, state.get_node_id()),
```

Comment:
Do we need the `cast` here because `get_node_id` returns an `Optional[int]`? We could think about changing the return type to `int` to avoid these (and raise an exception if the `node_id` wasn't set before)

Reply:
Yes, the `cast` is precisely because of that. Sure, makes sense to change it.